cmd/soroban-rpc/internal/db/db.go

```diff
@@ -130,21 +137,78 @@ func getLatestLedgerSequence(ctx context.Context, q db.SessionInterface, cache *
 	return result, nil
 }
 
+type ReadWriterMetrics struct {
+	TxIngestDuration, TxSqlDuration, TxCount prometheus.Observer
+}
+
 type readWriter struct {
+	log                   *log.Entry
 	db                    *DB
 	maxBatchSize          int
 	ledgerRetentionWindow uint32
+	passphrase            string
+
+	metrics *ReadWriterMetrics
+}
+
+// NewReadWriter constructs a new readWriter instance and configures the size of
+// ledger entry batches when writing ledger entries and the retention window for
+// how many historical ledgers are recorded in the database, hooking up
+// metrics for various DB ops.
+func NewReadWriterWithMetrics(
```

Comment:
I think we should only have one public constructor for ReadWriter because we probably never want to construct an instance without metrics. I would remove the other constructor (`NewReadWriter()`) and rename this one to `NewReadWriter()`. For tests, you can call `interfaces.MakeNoOpDeamon()` to make a testing daemon stub.

Reply:
sure, fair enough! done in 9fc64a9